docs/model_configuration.rst

```diff
@@ -158,6 +158,14 @@ minimal configuration above. Specifically:
   batching, the initial batch dimension must be variable-size for all inputs
   and outputs.
 
+* :ref:`PyTorch TorchScript <section-pytorch-models>` models have an optional
+  output configuration in the model configuration file to support cases where
+  there are variable number and/or datatypes of output. Both the input and
+  outputs in the configuration must follow a specific naming convention i.e.
+  "\<name\>__\<index\>". Where <index> is the index of the corresponding input or
+  output. If the model supports variable size input tensors, their dimensions
+  in the configuration must be -1 for all variable-size inputs and outputs.
```

Comment:
Is this true? For other model types the model configuration doesn't need to be -1 if the underlying model is variable sized. It is ok for the model config to require a specific size even if the model itself could handle other sizes.

Reply:
The provider_utils.cc and provider.cc perform a check while creating the input/output map. 

provider_utils has a check to see if the number of bytes in the request is equal to the number of bytes expected and throws an error otherwise. The -1 is to inform that it is a variable size input.

Similarly provider requires a check to verify the o/p shape is the same as the expected shape and the 1 is to inform that it is a variable size o/p. 

This is not native to libtorch and should be consistent with all other backends.

Reply:
Yes. But that is not what you said in your description. You can have a model that supports variable size in one or more dimension of one or more input or output tensors, but the model configuration doesn't need to be specified as -1 in (any or all of) those dimensions. That is, it is ok for the model configuration to be more restrictive then the actual model itself. The last paragraph in section "Model Configuration" says this. So I don't think you actually need to say anything specific for pytorch at all.

Reply:
Ok. Will remove this